File: vue-client/src/schemas/BotResults.ts
class BotResultsForBots {
  public id: number = 0;
  public rank: number = 0;
  public survival: number = 0;
  public lastSurvivorBonus: number = 0;
  public bulletDamage: number = 0;
  public bulletKillBonus: number = 0;
  public ramDamage: number = 0;
  public ramKillBonus: number = 0;
  public totalScore: number = 0;
  public firstPlaces: number = 0;
  public secondPlaces: number = 0;
  public thirdPlaces: number = 0;
}

export class BotResultsForObservers extends BotResultsForBots {
  public name: string = "";
  public version: string = "";
}

File: gui-app/src/main/kotlin/dev/robocode/tankroyale/gui/model/BotResults.kt
package dev.robocode.tankroyale.gui.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
@SerialName("BotResultsForObserver")
data class BotResults(
    val id: Int,
    val name: String,
    val version: String,
    val rank: Int,
    val survival: Int,
    val lastSurvivorBonus: Int,
    val bulletDamage: Int,
    val bulletKillBonus: Int,
    val ramDamage: Int,
    val ramKillBonus: Int,
    val totalScore: Int,
    val firstPlaces: Int,
    val secondPlaces: Int,
    val thirdPlaces: Int
)
